builder/rootfs/bin/boot

```diff
@@ -74,6 +88,27 @@ done
 docker build -t deis/slugbuilder /usr/local/src/slugbuilder/
 docker build -t deis/slugrunner /usr/local/src/slugrunner/
 
+function gen_host_keys {
+	# generate the keys, then set them up in etcd if not already set
+	/usr/bin/ssh-keygen -A
```

Comment:
It's not really about saving milliseconds, but I think it'd make the code a little more correct and easy to follow. Remember before that we've seen race conditions where an etcd key is set and then is attempted to be read immediately following (setting and using the slugbuilder/slugrunner keys). Let's do a sanity check before we generate new keys, and if they exist in etcd, just use them.